src/lib/Permit.sol

```diff
@@ -0,0 +1,13 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.13;
+
+/// @dev EIP-2612 Permit
+struct Permit {
```

Comment:
What is a Permit?  Can you add some comments explaining how a Permit relates to Agreements and Criteria?